Makefile

```diff
@@ -60,7 +58,22 @@ ifndef PIP_OPTIONS
 	PIP_OPTIONS :=
 endif
 
-ifneq ($(INCLUDE_TESTS_IN_COVERAGE),)
+# NOTE: We only run coverage on master and version branches and not on pull requests since
+# it has a big performance overhead and is very slow.
+ifeq ($(TRAVIS_PULL_REQUEST),false)
+	ENABLE_COVERAGE := yes
```

Comment:
LGTM, thanks :+1: 

But as discussed on Slack, I will try to move this step (setting up the environment variable) to `travis.yml` file.